src/domains/MainnetDomain.sol

```diff
@@ -13,9 +13,13 @@
 //
 // You should have received a copy of the GNU Affero General Public License
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
-pragma solidity 0.8.0;
+pragma solidity >=0.8.0;
 
-import "./IntegrationTest.t.sol";
+import {Domain} from "./Domain.sol";
 
-contract IntegrationTest_0_8_0 is IntegrationTest {
+contract MainnetDomain is Domain {
+
+    constructor(string memory name) Domain(name) {
```

Comment:
Yeah originally I had that, but the issue is there may actually be two instances of the same domain type. Arbitrum refers to the type of L2, but there are two instances of this live today - Arbitrum One and Arbitrum Nova. I think it's better to allow the user to define their own environmental variables in the foundry.toml file.

Additionally I want this to be able to deal with testnets as well.

Reply:
So you're thinking we would eventually have something like:
```solidity
new ArbitrumDomain("arbitrum-nova");
new ArbitrumDomain("arbitrum-one");
```

Testnets are tricky, ultimately since the contract is called `MainnetDomain` I thought we would be putting our eth goerli rpcs into `mainnet`, if not, instead of having `MainnetDomain("goerli")` maybe we change this contract naming to `L1Domain`

Reply:
Yes exactly.

Re: mainnet yeah I wasn't super sold on that naming either because of Goerli which supports Arbitrum and Optimism testnets. I'm also trying to avoid the L1 / L2 language as we may later introduce support for sidechains like Polygon PoS. Something to think on.

Reply:
ahh ok, then I see your point and passing the strings makes sense.

For testnet/mainnet: maybe this changes to `MainDomain` or `BaseDomain`? then it is less confusing between mainnet and testnets?

Reply:
I like `BaseDomain`.

Reply:
Discussed in chat to use "Root"